src/pages/home/report/PureReportActionItem.tsx

```diff
@@ -0,0 +1,1113 @@
+                    isHovered={hovered}
+                    contextMenuAnchor={popoverAnchorRef.current}
+                    containerStyles={displayAsGroup ? [] : [styles.mt2]}
+                    checkIfContextMenuActive={toggleContextMenuFromActiveReportAction}
+                />
+            );
+        } else if (action.actionName === CONST.REPORT.ACTIONS.TYPE.REPORT_PREVIEW) {
+            children = ReportUtils.isClosedExpenseReportWithNoExpenses(iouReport) ? (
+                <RenderHTML html={`<comment>${translate('parentReportAction.deletedReport')}</comment>`} />
+            ) : (
+                <ReportPreview
+                    iouReportID={ReportActionsUtils.getIOUReportIDFromReportActionPreview(action)}
+                    chatReportID={reportID}
+                    policyID={report?.policyID ?? '-1'}
+                    containerStyles={displayAsGroup ? [] : [styles.mt2]}
+                    action={action}
+                    isHovered={hovered}
+                    contextMenuAnchor={popoverAnchorRef.current}
+                    checkIfContextMenuActive={toggleContextMenuFromActiveReportAction}
+                    onPaymentOptionsShow={() => setIsPaymentMethodPopoverActive(true)}
+                    onPaymentOptionsHide={() => setIsPaymentMethodPopoverActive(false)}
+                    isWhisper={isWhisper}
+                />
+            );
+        } else if (ReportActionsUtils.isTaskAction(action)) {
+            children = <TaskAction action={action} />;
+        } else if (ReportActionsUtils.isCreatedTaskReportAction(action)) {
+            children = (
+                <ShowContextMenuContext.Provider value={contextValue}>
+                    <TaskPreview
+                        style={displayAsGroup ? [] : [styles.mt1]}
+                        taskReportID={ReportActionsUtils.isAddCommentAction(action) ? ReportActionsUtils.getOriginalMessage(action)?.taskReportID?.toString() ?? '-1' : '-1'}
+                        chatReportID={reportID}
+                        action={action}
+                        isHovered={hovered}
+                        contextMenuAnchor={popoverAnchorRef.current}
+                        checkIfContextMenuActive={toggleContextMenuFromActiveReportAction}
+                        policyID={report?.policyID ?? '-1'}
+                    />
+                </ShowContextMenuContext.Provider>
+            );
+        } else if (ReportActionsUtils.isReimbursementQueuedAction(action)) {
+            const linkedReport = ReportUtils.isChatThread(report) ? parentReport : report;
+            const submitterDisplayName = PersonalDetailsUtils.getDisplayNameOrDefault(personalDetails?.[linkedReport?.ownerAccountID ?? -1] ?? {});
+            const paymentType = ReportActionsUtils.getOriginalMessage(action)?.paymentType ?? '';
+
+            const missingPaymentMethod = ReportUtils.getIndicatedMissingPaymentMethod(userWallet, linkedReport?.reportID ?? '-1', action);
```

Comment:
Here too `getIndicatedMissingPaymentMethod > isCurrentUserSubmitter > getAllReports` and `getIndicatedMissingPaymentMethod > isSettled > getAllReports` and `hasCreditBankAccount`

Reply:
done